hack/central-repo/upload-plugins.sh

```diff
@@ -0,0 +1,140 @@
+    dry_run=echo
+    shift
+fi
+
+fast=off
+if [ "$1" = "--fast" ]; then
+    fast=on
+    shift
+fi
+
+if [ $# -eq 0 ] || [[ $1 == "-"* ]]; then
+    usage
+fi
+
+content_image=$1
+repoBasePath=$(dirname $content_image)
+database=/tmp/central.db
+publisher="vmware/tkg"
+
+# Create db table
+cat << EOF | sqlite3 -batch $database
+CREATE TABLE IF NOT EXISTS "PluginBinaries" (
+		"PluginName"         TEXT NOT NULL,
+		"Target"             TEXT NOT NULL,
+		"RecommendedVersion" TEXT NOT NULL,
+		"Version"            TEXT NOT NULL,
+		"Hidden"             INTEGER NOT NULL,
+		"Description"        TEXT NOT NULL,
+		"Publisher"          TEXT NOT NULL,
+		"Vendor"             TEXT NOT NULL,
+		"OS"                 TEXT NOT NULL,
+		"Architecture"       TEXT NOT NULL,
+		"Digest"             TEXT NOT NULL,
+		"URI"                TEXT NOT NULL,
+		PRIMARY KEY("PluginName", "Target", "Version", "OS", "Architecture")
+	);
+EOF
+
+addPlugin() {
+    name=$1
+    target=$2
+
+    tmpPluginPhase1="/tmp/fakeplugin1.sh"
+    tmpPluginPhase2="/tmp/fakeplugin2.sh"
+
+    # Start preparing the plugin file with the name and target.
+    # Start here to avoir repeating in the loop.
+    sed -e "s/__NAME__/$name/" -e "s/__TARGET__/$target/" $ROOT_DIR/fakeplugin.sh > $tmpPluginPhase1
+
+    # Define 10 versions for the plugin
+    for v in {0..9}; do
+        version="v$v.$v.$v"
+
+        # Put printout to show progress
+        echo "Inserting $name version $version for target $target"
+
+        # Create the plugin file with the correct version
+        sed -e "s/__VERSION__/$version/" $tmpPluginPhase1 > $tmpPluginPhase2
+        digest=$(sha256sum $tmpPluginPhase2 | cut -f1 -d' ')
+
+        for os in darwin linux windows; do
+            for arch in amd64 arm64; do
+                if [ $arch = arm64 ] && [ $os != darwin ]; then
+                    # Only support darwin with arm64 for now
+                    continue
+                fi
+
+                image_path=$publisher/$os/$arch/$target/$name
+
+                sql_cmd="INSERT INTO PluginBinaries VALUES('$name','$target','v9.9.9','$version','FALSE','Desc for $name','TKG','VMware','$os','$arch','$digest','$image_path:$version');"
+                if [ "$dry_run" = "echo" ]; then
+                    echo $sql_cmd 
+                else 
+                    echo $sql_cmd | sqlite3 -batch $database
+                fi
+
+                # For efficiency, only push the plugin binaries of the two latest versions for 4 plugins
+                # The plugins are: twotargets (for both targets), kubeplugin, tmcplugin 
+                if [ $name = "twotargets" ] || [ $name = "kubeplugin" ] || [ $name = "tmcplugin" ]; then
+                    if [ $version = "v8.8.8" ] || [ $version = "v9.9.9" ]; then
+                        ${dry_run} imgpkg push -i $repoBasePath/$image_path:$version -f $tmpPluginPhase2 --registry-insecure
+                    fi
+                fi
```

Comment:
We should also document this in the README specifying only these plugin binaries will be available in the central repository. Other plugins are added to the database for testing but plugin binaries are not available within this local test central repository.

Also, the PR description mentions running `imgpkg pull -i localhost:9876/tanzu-cli/plugins/vmware/tkg/darwin/arm64/tmc/plugin0:v9.9.9 -o tmp2` command to fetch plugin binary but binary for `plugin0` doesn't exist in the central repository and command fails with an error. PR description needs to be updated to specify the plugin binary which exists in the local test central repository.

Reply:
> We should also document this in the README specifying only these plugin binaries will be available in the central repository. Other plugins are added to the database for testing but plugin binaries are not available within this local test central repository.

😉  https://github.com/vmware-tanzu/tanzu-cli/pull/41/files#diff-3f8fe7016a74b43b2511fe3745969ec27a165a0e396225e8da9fd98365f4fd48R41-R44

BTW, this approach is a little awkward because not all plugins are available to install.
But it does make generation much faster and the `registry-content.bz2` git file much smaller.
Do you think it makes using the test central repo too difficult?

> Also, the PR description mentions running `imgpkg pull -i localhost:9876/tanzu-cli/plugins/vmware/tkg/darwin/arm64/tmc/plugin0:v9.9.9 -o tmp2` command to fetch plugin binary but binary for `plugin0` doesn't exist in the central repository and command fails with an error. PR description needs to be updated to specify the plugin binary which exists in the local test central repository.

Woops!  Updated.  Thanks.